packages/grafana-ui/src/components/IconButton/IconButton.tsx

```diff
@@ -46,12 +46,19 @@ export const IconButton = React.forwardRef<HTMLButtonElement, Props>(
     ref
   ) => {
     const theme = useTheme2();
-    const styles = getStyles(theme, size, variant);
+    let limitedIconSize: IconSize = size;
+
+    // very large icons (xl to xxxl) are unified to size xl
+    if (size === 'xxl' || size === 'xxxl') {
+      limitedIconSize = 'xl';
+    }
```

Comment:
I think if we are doing this, `limitedIconSize` is no longer of type `IconSize`, because it doesn't accept two of those sizes. Can we add a comment on the `size` prop regarding xxl and xxxl not being accepted anymore as well? Otherwise this has confusing results for someone using it 😅